File: Classes/EVEOnlineAPI/EVE/EVEErrorList.m
//
//  EVEErrorList.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/17/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEErrorList.h"


@implementation EVEErrorListItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"errorCode":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"errorText":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	
	return scheme;
}

@end


@implementation EVEErrorList

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"errors":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEErrorListItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEErrorList.swift
//
//  EVEErrorList.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEErrorListItem: EVEObject {
	public var errorCode: Int = 0
	public var errorText: String = ""
	
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"errorCode":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"errorText":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}

public class EVEErrorList: EVEResult {
	public var errors: [EVEErrorListItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"errors":EVESchemeElementType.Rowset(elementName: nil, type: EVEErrorListItem.self, transformer: nil)
		]
	}
}

